tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
+using FileExplorerSourceControlIntegration;
+using LibGit2Sharp;
+using Microsoft.UI.Xaml.Media.Animation;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Storage;
+
+namespace DevHome.FileExplorerSourceControlIntegration.Services;
+
+public class RepositoryTracking
+{
+    public RepoStoreOptions? RepoStoreOptions
+    {
+        get; set;
+    }
+
+    public enum RepositoryChange
+    {
+        Added,
+        Removed,
+    }
+
+    private readonly FileService fileService;
+
+    private readonly object trackRepoLock = new();
+
+    private Dictionary<string, string> TrackedRepositories { get; set; } = new Dictionary<string, string>();
+
+    private readonly Serilog.ILogger log = Log.ForContext("SourceContext", nameof(RepositoryTracking));
+
+    public event Windows.Foundation.TypedEventHandler<string, RepositoryChange>? RepositoryChanged;
+
+    public RepositoryTracking(string? path)
+    {
+        if (RuntimeHelper.IsMSIX && path == null)
+        {
+            RepoStoreOptions = new RepoStoreOptions
+            {
+                RepoStoreFolderPath = ApplicationData.Current.LocalFolder.Path,
+            };
+            log.Debug("Repo Store for File Explorer Integration created under ApplicationData");
+        }
+        else
+        {
+            RepoStoreOptions = new RepoStoreOptions

```

Comment:
This code path will light up if `IsMSIX` is false and path is null.  Is this okay?

Reply:
Fixed this in latest commit. Thanks for the catch here!